src/pages/MainPage.vue

```diff
@@ -426,15 +425,14 @@ export default {
       }
     },
     setActiveRoad: function () {
-      const roadHash = window.location.hash;
-      if (roadHash.length && roadHash.substring(0, 7) === '#/#road') {
-        const roadRequested = roadHash.substring(7);
+      if (this.roads.hasOwnProperty(this.$route.params.road)) {
+        const roadRequested = this.$route.params.road;
         if (roadRequested in this.roads) {
-          this.$store.commit('setActiveRoad', roadHash.substring(7));
+          this.$store.commit('setActiveRoad', roadRequested);
           return true;
         }
       }
-      window.location.hash = '#/#road' + this.activeRoad;
+      this.$router.push({ path: `/road/${this.activeRoad}` });
```

Comment:
I've also seen it done like
```js
router.push({ name: 'road', params: { road: this.activeRoad } })
```
for named routes, which lets us change the path name without changing all pushes to it. Seems more extensible, but idk the arguments against it.

Reply:
I used the method above because the `routes` we currently have aren't actually named and we couldn't use the `named route` method. If we want, we could go back and set an explicit name for each `route` which would allow us to do this.